internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -101,78 +103,108 @@ func (r *ClusterDeploymentsReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, nil
 	}
 
+	clusterInstall := &hiveext.AgentClusterInstall{}
+	if clusterDeployment.Spec.ClusterInstallRef == nil {
+		r.Log.Infof("AgentClusterInstall not set for ClusterDeployment %s", clusterDeployment.Name)
+		return ctrl.Result{}, nil
+	}
+
+	aciName := clusterDeployment.Spec.ClusterInstallRef.Name
+	err = r.Get(ctx,
+		types.NamespacedName{
+			Namespace: clusterDeployment.Namespace,
+			Name:      aciName,
+		},
+		clusterInstall)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			r.Log.WithField("AgentClusterInstall", aciName).Infof("AgentClusterInstall does not exist for ClusterDeployment %s", clusterDeployment.Name)
+			return ctrl.Result{}, nil
+		}
+		r.Log.WithError(err).Errorf("Failed to get AgentClusterInstall %s", aciName)
+		return ctrl.Result{Requeue: true}, err
+	}
```

Comment:
Could I make a request for this location, in the enhancement we talked about having an owner ref from CD to CI, so deleting the CD would trickle down and delete the CI automatically. I went to implement this in Hive so it would apply to anyone who implemented a ClusterInstall, but realized we don't have a great way to update types we don't explicitly know right now, and also that I'm not sure the hive controllers should be writing to a CI, just as we don't want CI controllers writing to a CD. If we set this owner ref in the respective CI controllers like this one, then the authors of those controllers see the functionality and are less likely to be surprised by it. Maybe some wouldn't want that owner ref. Essentially asking for lifecycle of the CI to be managed in controllers like this one, including finalizers and owner references.

The change would be something like: https://github.com/openshift/hive/blob/master/pkg/controller/clusterdeployment/clusterdeployment_controller.go#L1852, and then return if it made a change and re-reconcile.

Does this seem ok?

Reply:
You want an ownerref where CD is the owner of CI?

Reply:
Yes exactly.

Reply:
Will handle in another PR. Opened Jira ticket: MGMT-6420